### DISCUSSION TITLE
如何使用vscode来加速RT-Thread开发 | How to use vscode to accelerate RT Thread development.

### DISCUSSION THREAD

**Discussion by BernardXiong** [RT-Thread/rt-thread#8046] [Ideas]
目前大家基本上都会使用VSCode，当然也包括把VSCode用于RT-Thread的开发。所以这里也列了一些，如何来使用VSCode加速RT-Thread的开发，这里想到了几点：

* 自动安装环境
  * ARM Cortex-M类
  * RISC-V类
  * 可以自动安装工具链，或一些工具；
* 生成工程
  * 需要更多以工程为中心的方式生成，而不是RT-Thread BSP；
  * 可以以链接方式链接到实际的RT-Thread目录（包括Windows环境，目前也已经支持符号链接了(对权限似乎有要求)）
  * 支持RT-Thread Smart应用程序创建
* 应该更多以命令/cmd的方式来完成各类工作；
* 支持RT-Thread配置及软件包功能；
* 比较好的支持到qemu及一些测试相关的自动化动作；
* 也可以集成一些有用的VSCode插件
  * [Embedded Tools](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-embedded-tools)
  * [Serial Monitor](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-serial-monitor)
  * [Cortex-Debug](https://marketplace.visualstudio.com/items?itemName=marus25.cortex-debug)
  * [RTOS Views](https://marketplace.visualstudio.com/items?itemName=mcu-debug.rtos-views)

**Comment by helloeagleyang** [comment]
建议支持直接scons --target=vscode 这种方式来生成用于RT-Thread Studio 插件的工程，并且能把toolchain什么的直接填好，方便用户直接编译调试，这样能解决目前RT-Thread Studio只支持windows操作系统这个限制。

**Reply by BernardXiong** [reply]
别，既然是vscode，那么想法就是完全vscode相关的，而不是和studio相关。

**Comment by BernardXiong** [comment]
可以考虑引入一些辅助性工具类型的，例如类似env的存在。这样可以和python脚本结合起来，同时不需要和scons结合起来的方式，来提供一定的功能，例如：
* 创建一个工程

**Comment by BernardXiong** [comment]
* 自动安装环境
  * ARM Cortex-M类
  * RISC-V类
  * 可以自动安装工具链，或一些工具；

⟹ 可以是一类软件包：
* 每个软件包可以携带份描述文件
  * 描述这个软件包的信息，介绍；
  * 一些变量或环境变量，这些变量都应该是安装、展开后目录相关的。当安装后，即可引用这些变量。
* 可以有个总的变量，定义这些软件包的根目录，是默认目录或统一安装到指定目录
